qiskit/chemistry/drivers/fcidumpd/fcidumpdriver.py

```diff
@@ -0,0 +1,104 @@
+        Computer Physics Communications, Volume 54, Issue 1, 1989, Pages 75-83,
+        ISSN 0010-4655, https://doi.org/10.1016/0010-4655(89)90033-7.
+    """
+
+    def __init__(self, fcidump_input: str, atoms: list = None) -> None:
+        """
+        Initializer
+
+        Args:
+            fcidump_input: path to the FCIDump file
+            atoms (optional): Allows to specify the atom list of the molecule. If it is provided,
+            the created QMolecule instance will permit frozen core Hamiltonians. This list must
+            consist of valid atom symbols.
+
+        Raises:
+            QiskitChemistryError: invalid input
+        """
+        super().__init__()
+
+        if not isinstance(fcidump_input, str):
+            raise QiskitChemistryError("Invalid input for FCIDumpDriver '{}'".format(fcidump_input))
+        self._fcidump_input = fcidump_input
+
+        if atoms is not None and not isinstance(atoms, list) \
+                and not all([sym in QMolecule.symbols for sym in atoms]):
+            raise QiskitChemistryError("Invalid input for FCIDumpDriver '{}'".format(atoms))
+        self.atoms = atoms
+
+    def run(self) -> QMolecule:
+        """
+        Constructs a QMolecule instance out of a FCIDump file.
+
+        Returns:
+            QMolecule: a QMolecule instance populated with a minimal set of required data
+        """
+        fcidump_data = parse(self._fcidump_input)
+
+        q_mol = QMolecule()
+
+        q_mol.nuclear_repulsion_energy = fcidump_data.get('ecore', float('NaN'))
+        q_mol.num_orbitals = fcidump_data.get('NORB', float('NaN'))
+        q_mol.multiplicity = fcidump_data.get('MS2', 0) + 1
+        q_mol.num_beta = (fcidump_data.get('NELEC', float('NaN')) - (q_mol.multiplicity - 1)) // 2
+        q_mol.num_alpha = fcidump_data.get('NELEC', float('NaN')) - q_mol.num_beta
+        if self.atoms is not None:
+            q_mol.num_atoms = len(self.atoms)
+            q_mol.atom_symbol = self.atoms
+
+        q_mol.mo_onee_ints = fcidump_data.get('hij', None)
+        q_mol.mo_onee_ints_b = fcidump_data.get('hij_b', None)
+        q_mol.mo_eri_ints = fcidump_data.get('hijkl', None)
+        q_mol.mo_eri_ints_bb = fcidump_data.get('hijkl_bb', None)
+        q_mol.mo_eri_ints_ba = fcidump_data.get('hijkl_ba', None)
+
+        return q_mol
+
+    @staticmethod
+    def dump(q_mol: QMolecule, outpath: str, orbsym: list = None, isym: int = 1) -> None:
+        """
+        Convenience method to produce an FCIDump output file
+
+        Args:
+            outpath (str): path to the output file
+            q_mol (QMolecule): QMolecule data to be dumped. It is assumed that the HF energy stored
+            in this QMolecule instance contains the inactive core energy.
+            orbsym (optional): list of spatial symmetries of the orbitals
+            isym (optional): spatial symmetry of the wave function. Defaults to 1.
+        """
```

Comment:
* I'm not sure whether `orbsym` is actually a list of int
* If the default value is already present in the signature, you don't necessarily have to mention it in the docstring again

```suggestion
    def dump(q_mol: QMolecule, outpath: str, orbsym: Optional[List[int]] = None, isym: int = 1) -> None:
        """Convenience method to produce an FCIDump output file

        Args:
            outpath: Path to the output file.
            q_mol: QMolecule data to be dumped. It is assumed that the HF energy stored
                in this QMolecule instance contains the inactive core energy.
            orbsym: A list of spatial symmetries of the orbitals.
            isym: The spatial symmetry of the wave function. 
        """
```